### ISSUE TITLE
SecurityException in Utils.getDeviceNetworkType()

### ISSUE THREAD

**Issue by rizalferdian** [CleverTap/clevertap-android-sdk#147]
There were many crashes reported related to `Utils.java` class in our apps. we use clevertap v4.0.0

**Stack Trace**
```
Fatal Exception: java.lang.SecurityException: Neither user 10156 nor current process has android.permission.MODIFY_PHONE_STATE.
       at android.os.Parcel.createException(Parcel.java:1966)
       at android.os.Parcel.readException(Parcel.java:1934)
       at android.os.Parcel.readException(Parcel.java:1884)
       at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkType(ITelephony.java:4233)
       at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2747)
       at com.clevertap.android.sdk.Utils.getDeviceNetworkType(Utils.java:282)
       at com.clevertap.android.sdk.DeviceInfo$DeviceCachedInfo.getNetworkType(DeviceInfo.java:172)
       at com.clevertap.android.sdk.DeviceInfo$DeviceCachedInfo.<init>(DeviceInfo.java:67)
       at com.clevertap.android.sdk.DeviceInfo.getDeviceCachedInfo(DeviceInfo.java:518)
       at com.clevertap.android.sdk.DeviceInfo.access$100(DeviceInfo.java:21)
       at com.clevertap.android.sdk.DeviceInfo$1.run(DeviceInfo.java:264)
       at java.lang.Thread.run(Thread.java:764)
```
Currently this issue only happen when the app is on the background with specific samsung devices (SM-T380, SM-T385)

**Comment by rizalferdian** [comment]
@darshanclevertap the `targetSdkVersion 29` and the `compileSdkVersion 16`

**Comment by darshanclevertap** [comment]
@rizalferdian Are you also using the CleverTap Xiaomi Push SDK?

**Comment by rizalferdian** [comment]
@darshanclevertap no i'm not

**Comment by darshanclevertap** [comment]
@rizalferdian Can you check with `targetSdkVersion` as 30 and update the CleverTap SDK version to 4.0.2?

**Comment by darshanclevertap** [comment]
@rizalferdian Are you still facing this issue?

**Comment by rizalferdian** [comment]
@darshanclevertap Currently I cannot reproduce the bug myself, if we have to change the `targetSdkVersion` to 30 I think it will take a while before it goes to production.

**Comment by piyush-kukadiya** [comment]
@rizalferdian If crashes are high then please downgrade SDK to `3.9.1`. We will be fixing this in next release.

**Comment by rizalferdian** [comment]
Okay thank you

**Comment by kumaratulismu** [comment]
@rizalferdian We have fixed this in our latest release v4.0.3. 
Closing this.

**Comment by Prashant-Sarin** [comment]
We observe this again on the latest version (4.2.0). Please help @rizalferdian
